binding.cc

```diff
@@ -390,8 +390,7 @@ namespace zmq {
 
   void
   Socket::UV_PollCallback(uv_poll_t* handle, int status, int events) {
-    assert(status == 0);
-
+    if (status != 0) NanThrowError(ExceptionFromError());
```

Comment:
That's how I've always interpreted it too, and I think that's what the spec will tell you.